packages/expo-screen-orientation/ios/EXScreenOrientation/EXScreenOrientationUtilities.m

```diff
@@ -0,0 +1,170 @@
+// Copyright 2019-present 650 Industries. All rights reserved.
+
+#import <Foundation/Foundation.h>
+#import <EXScreenOrientation/EXScreenOrientationUtilities.h>
+
+#import <sys/utsname.h>
+
+@implementation EXScreenOrientationUtilities
+
+# pragma mark - helpers
+
++ (BOOL)doesSupportOrientationMask:(UIInterfaceOrientationMask)orientationMask
+{
+  if ((UIInterfaceOrientationMaskPortraitUpsideDown & orientationMask) // UIInterfaceOrientationMaskPortraitUpsideDown is part of orientationMask
+      && [self doesDeviceHaveNotch])
+  {
+    // device does not support UIInterfaceOrientationMaskPortraitUpsideDown and it was requested via orientationMask
+    return false;
+  }
+  
+  return true;
+}
+
++ (BOOL)doesDeviceHaveNotch {
+  if (@available(iOS 11.0, *)) {
+    __block BOOL result = false;
+    dispatch_semaphore_t sem = dispatch_semaphore_create(0);
+    dispatch_async(dispatch_get_main_queue(), ^{
+      result = ([[[UIApplication sharedApplication] delegate] window].safeAreaInsets.top ?: 0.0) > 20.0;
+      dispatch_semaphore_signal(sem);
+    });
+    dispatch_semaphore_wait(sem, DISPATCH_TIME_FOREVER);
+    return result;
+  }
+  
+  return false;
+}
+
++ (UIInterfaceOrientationMask)maskFromOrientation:(UIInterfaceOrientation)orientation
+{
+  switch (orientation) {
+    case UIInterfaceOrientationPortrait:
+      return UIInterfaceOrientationMaskPortrait;
+    case UIInterfaceOrientationPortraitUpsideDown:
+      return UIInterfaceOrientationMaskPortraitUpsideDown;
+    case UIInterfaceOrientationLandscapeLeft:
+        return UIInterfaceOrientationMaskLandscapeLeft;
+    case UIInterfaceOrientationLandscapeRight:
+      return UIInterfaceOrientationMaskLandscapeRight;
+    default:
+      return 0;
+  }
+}
+
++ (UIInterfaceOrientation)UIDeviceOrientationToUIInterfaceOrientation:(UIDeviceOrientation)deviceOrientation
```

Comment:
When we move away from depending on `UIDeviceOrientation` this could be removed.